vmdb/app/helpers/ems_cloud_helper/textual_summary.rb

```diff
@@ -46,7 +46,7 @@ def textual_type
   end
 
   def textual_port
-    ["openstack", "openstack_infra", "rhevm"].include?(@ems.emstype) ? {:label => "API Port", :value => @ems.port} : nil
+    @ems.class.supports_port? ? {:label => "API Port", :value => @ems.port} : nil
```

Comment:
Because it's not different per instance, i. e. there is no state.  But
having an instance level accessor makes sense.
On Jan 28, 2015 2:31 PM, "Oleg Barenboim" <notifications@github.com> wrote:

> In vmdb/app/helpers/ems_cloud_helper/textual_summary.rb
> <https://github.com/ManageIQ/manageiq/pull/1476#discussion_r23715324>:
>
> > @@ -46,7 +46,7 @@ def textual_type
> >    end
> >
> >    def textual_port
> > -    ["openstack", "openstack_infra", "rhevm"].include?(@ems.emstype) ? {:label => "API Port", :value => @ems.port} : nil
> > +    @ems.class.supports_port? ? {:label => "API Port", :value => @ems.port} : nil
>
> Now that I reviewed all the callers of the supports_port? method, I am
> questioning why we have it at the class level at all? All the callers take
> an ems and do ems.class.supports_port?. That seems wrong.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ManageIQ/manageiq/pull/1476/files#r23715324>.
>